packages/react/src/table/TableScroller.tsx

```diff
@@ -209,11 +209,25 @@ export function TableScroller({ children }: TableScrollerProps) {
 	const hasScroll = thumbWidthRatio !== 1;
 
 	useEffect(() => {
-		setScrollerAriaLabel(
-			`Table ${
-				scrollerRef.current?.querySelector('caption')?.textContent || ''
-			}`
-		);
+		let ariaLabel = '';
+		const captionEl = scrollerRef.current?.querySelector('caption');
+
+		if (captionEl) {
+			ariaLabel = captionEl?.textContent || '';
+		} else {
+			const ariaLabelledbyTableEl = scrollerRef.current?.querySelector(
+				'table[aria-labelledby]'
+			);
+
+			if (ariaLabelledbyTableEl) {
+				ariaLabel =
+					document.getElementById(
+						ariaLabelledbyTableEl.getAttribute('aria-labelledby') || ''
+					)?.textContent || '';
+			}
+		}
+
+		setScrollerAriaLabel(`Table ${ariaLabel}`);
 	}, [scrollerRef]);
```

Comment:
Updates to `ref`s don't trigger re-renders so this doesn't work as likely intended.

Reply:
But it triggers the useEffect which does cause the re-render? It all seems to function?

Reply:
[React docs say it's fine to leave in](https://react.dev/learn/synchronizing-with-effects):

>React guarantees [you’ll always get the same object](https://react.dev/reference/react/useRef#returns) from the same useRef call on every render. It never changes, so it will never by itself cause the Effect to re-run. Therefore, it does not matter whether you include it or not. Including it is fine too:

Spoke offline about this ⬆️ and decided to remove.